presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -90,10 +91,13 @@ public ConnectorPageSource createPageSource(ConnectorTransactionHandle transacti
         HiveSplit hiveSplit = (HiveSplit) split;
         Path path = new Path(hiveSplit.getPath());
 
+        Configuration configuration = new Configuration(hdfsEnvironment.getConfiguration(new HdfsContext(session, hiveSplit.getDatabase(), hiveSplit.getTable()), path));
+        configuration.setBoolean(S3_SELECT_PUSHDOWN_ENABLED, hiveSplit.isS3SelectPushdownEnabled());
```

Comment:
We only want to use Hadoop `Configuration` for Hadoop APIs, as this effectively creates hidden state. Instead, since this is our code, let's add it as a parameter to `HiveRecordCursorProvider`.